hastory/src/Hastory/Storage.hs

```diff
@@ -0,0 +1,39 @@
+{-# LANGUAGE DeriveGeneric              #-}
+{-# LANGUAGE EmptyDataDecls             #-}
+{-# LANGUAGE FlexibleContexts           #-}
+{-# LANGUAGE GADTs                      #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE MultiParamTypeClasses      #-}
+{-# LANGUAGE OverloadedStrings          #-}
+{-# LANGUAGE QuasiQuotes                #-}
+{-# LANGUAGE TemplateHaskell            #-}
+{-# LANGUAGE TypeFamilies               #-}
+
+module Hastory.Storage where
+
+import Hastory.Storage.HastoryPath
+import Hastory.Storage.HastoryZonedTime
+
+import Prelude (String, Show)
+import Database.Persist.Sqlite ()
+import Database.Persist.TH (persistLowerCase, share, mkPersist, sqlSettings, mkMigrate)
+import GHC.Generics (Generic)
+import Data.Text (Text)
+
+-- data Entry = Entry
+--     { entryText :: Text
+--     , entryWorkingDir :: Path Abs Dir
+--     , entryDateTime :: ZonedTime
+--     , entryHostName :: String
+--     , entryUser :: String
+--     } deriving (Show, Generic)
+
+share [mkPersist sqlSettings, mkMigrate "migrateAll"] [persistLowerCase|
+Entry
+    text Text
+    workingDir HastoryPath
+    dateTime HastoryZonedTime
+    hostName String
```

Comment:
`String` will be problematic, because `T.unpack . T.pack` is not equivalent to `id`.
Best to use `Text` instead.

Reply:
I went with `Text`.